File: packages/jsii-pacmak/test/expected.java/org/jsii/tests/calculator/Polymorphism.java
package org.jsii.tests.calculator;
@org.jsii.Jsii(module = org.jsii.tests.calculator.$Module.class, fqn = "jsii$jsii_calc$.Polymorphism")
public class Polymorphism extends org.jsii.JsiiObject {
    protected Polymorphism(final org.jsii.JsiiObject.InitializationMode mode) {
        super(mode);
    }
    public Polymorphism() {
        super(org.jsii.JsiiObject.InitializationMode.Jsii);
        org.jsii.JsiiEngine.getInstance().createNewObject(this);
    }
    public java.lang.String sayHello(final org.jsii.tests.calculator.lib.IFriendly friendly) {
        return this.jsiiCall("sayHello", java.lang.String.class, friendly);
    }
}

File: packages/jsii-dotnet-runtime/test/AWS.Jsii.Tests.Calculator/Polymorphism.cs
using AWS.Jsii.Runtime.Deputy;
using AWS.Jsii.Tests.Calculator.Lib;

namespace AWS.Jsii.Tests.Calculator
{
    [JsiiClass("jsii-calc", "jsii$jsii_calc$.Polymorphism", "[]")]
    public class Polymorphism : DeputyBase
    {
        public Polymorphism(): base(new DeputyProps(new object[]{}))
        {
        }

        protected Polymorphism(ByRefValue reference): base(reference)
        {
        }

        protected Polymorphism(DeputyProps props): base(props)
        {
        }

        [JsiiMethod("sayHello", "{\"primitive\":\"string\"}", "[{\"name\":\"friendly\",\"type\":{\"fqn\":\"jsii$jsii_calc_lib$.IFriendly\"}}]")]
        public virtual string SayHello(IIFriendly friendly)
        {
            return InvokeMethod<string>(new object[]{friendly});
        }
    }
}